### PR TITLE
chore(refactor): Move pipelines to compliance operator packages

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description

Move the compliance operator related pipelines from the central/sensor/service/pipeline package to central/complianceoperator/pipelines and central/complianceoperator/v2/pipelines .
Reasons:

 - The `pipeline` package is not scalable, it grows with every new added resource we synchronize in Central
 - Navigation and exploration of the source code is hard. One cannot find the logic which upserts new resources easily without knowing about the pipelines. 
 - Code belonging to a domain/use-case (e.g. complianceoperator) can be found in one location.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

 - CI

### CODE REVIEW

File: central/complianceoperator/pipelines/complianceoperator/complianceoperatorinfo/pipeline.go
Comment (dashrews78) [inline central/complianceoperator/pipelines/complianceoperator/complianceoperatorinfo/pipeline.go:1]: This is actually a V2 pipeline which sort of illustrates why this change is a good idea.

So move this to v2 and I would also remove the second `complianceoperator` directory from this one.

**Review by dashrews78 — CHANGES_REQUESTED** [review]
Move the complianceoperatorinfo to v2.

File: central/complianceoperator/pipelines/complianceoperator/complianceoperatorinfo/pipeline.go
Comment (SimonBaeumer) [reply central/complianceoperator/pipelines/complianceoperator/complianceoperatorinfo/pipeline.go:1]: Great, already uncovered an issue. Fixed it.

**Review by lvalerom — COMMENTED** [review]
Looks good! Would it make sense to have the pipelines inside of each resource folder similarly to the `datastore` and `service` packages (e.g. `central/complianceopeartor/v2/rules/pipeline`). It would look something like this:

- central
    - complianceoperator
        - v2
            - rules
                - datastore
                - pipeline
                - store
            - profiles
                - datastore
                - pipeline
                - service
                - store

**Comment by SimonBaeumer** [comment]
> Looks good! Would it make sense to have the pipelines inside of each resource folder similarly to the `datastore` and `service` packages (e.g. `central/complianceopeartor/v2/rules/pipeline`). It would look something like this:
> 
> * central
>   
>   * complianceoperator
>     
>     * v2
>       
>       * rules
>         
>         * datastore
>         * pipeline
>         * store
>       * profiles
>         
>         * datastore
>         * pipeline
>         * service
>         * store

Good observation. 
I've tried but did not liked the outcome. First not all resource could be mapped to one package, the `pipelines` directory would persist. 
It's to me a bit too make it worse and split the pipelines too much. With a bit more experience we can move them later into the respective packages.

**Comment by dashrews78** [comment]
> > Looks good! Would it make sense to have the pipelines inside of each resource folder similarly to the `datastore` and `service` packages (e.g. `central/complianceopeartor/v2/rules/pipeline`). It would look something like this:
> > 
> > * central
> >   
> >   * complianceoperator
> >     
> >     * v2
> >       
> >       * rules
> >         
> >         * datastore
> >         * pipeline
> >         * store
> >       * profiles
> >         
> >         * datastore
> >         * pipeline
> >         * service
> >         * store
> 
> Good observation. I've tried but did not liked the outcome. First not all resource could be mapped to one package, the `pipelines` directory would persist. It's to me a bit too make it worse and split the pipelines too much. With a bit more experience we can move them later into the respective packages.

I agree.  that's an interesting idea, but lets save that for a later and larger refactor

**Review by dashrews78 — APPROVED** [review]

**Pull request merged**